shell/browser/native_window_views.cc

```diff
@@ -235,10 +235,14 @@ NativeWindowViews::NativeWindowViews(const gin_helper::Dictionary& options,
   options.Get(options::kMinimizable, &minimizable_);
   options.Get(options::kMaximizable, &maximizable_);
 
-  // Transparent window must not have thick frame.
-  options.Get("thickFrame", &thick_frame_);
-  if (transparent())
+  // Transparent window does not have thick frame for default.
+  absl::optional<bool> thick_frame_optional;
+  options.GetOptional("thickFrame", &thick_frame_optional);
+  if (transparent() && !thick_frame_optional.has_value()) {
```

Comment:
I think transparent windows and WS_THICKFRAME are mutually exclusive, if a window is transparent it cannot have a frame and non-client area elements.

Another comment in this direction: https://source.chromium.org/chromium/chromium/src/+/main:ui/views/win/hwnd_message_handler.cc;l=967?q=hwnd_message_handler.cc&ss=chromium%2Fchromium%2Fsrc

Reply:
@brhenrique A `BrowserWindow` with arcylic window is not a transparent window technically, but it needs `transparent: true` option to work, because only when `transparent` is true it can have background color with alpha channel. The `vibrancy` option for macOS has a similar logic.

Reply:
Material backgrounds won't need `transparent: true` in order to work once [this fix is merged](https://github.com/electron/electron/pull/40395/files#diff-9a00c5b8d17617acce09892658a116c0f0f0d03c1cf7e4d59d8247c2ecfa911cR147), it is a bug from #39895.

Reply:
@brhenrique Have tested your branch and it's indeed the better solution. Nice work!